src/node_file.cc

```diff
@@ -2727,6 +2730,222 @@ static void Mkdtemp(const FunctionCallbackInfo<Value>& args) {
+
+  if (hostname.size() > 0) {
+    // If hostname is set, then we have a UNC path
+    // Pass the hostname through domainToUnicode just in case
+    // it is an IDN using punycode encoding. We do not need to worry
+    // about percent encoding because the URL parser will have
+    // already taken care of that for us. Note that this only
+    // causes IDNs with an appropriate `xn--` prefix to be decoded.
+    *result_file_path =
+        "\\\\" + ada::unicode::to_unicode(hostname) + decoded_pathname;
+
+    return true;
+  }
+
+  char letter = decoded_pathname[1] | 0x20;
+  char sep = decoded_pathname[2];
+
+  // a..z A..Z
+  if (letter < 'a' || letter > 'z' || sep != ':') {
+    env->isolate()->ThrowException(ERR_INVALID_FILE_URL_PATH(
+        env->isolate(), "File URL path must be absolute"));
+
+    return false;
+  }
+
+  result_file_path = decoded_pathname.substr(1);
+
+  return true;
+#else
+  std::string_view hostname = file_url.get_hostname();
+
+  if (hostname.size() > 0) {
+    std::string error_message =
+        std::string("File URL host must be \"localhost\" or empty on ") +
+        std::string(per_process::metadata.platform);
+    env->isolate()->ThrowException(
+        ERR_INVALID_FILE_URL_HOST(env->isolate(), error_message.c_str()));
+
+    return false;
+  }
+
+  for (size_t i = 0; (i + 2) < pathname.size(); i++) {
+    if (pathname[i] == '%' && pathname[i + 1] == '2' &&
+        (pathname[i + 2] | 0x20) == 102) {
+      env->isolate()->ThrowException(ERR_INVALID_FILE_URL_PATH(
+          env->isolate(),
+          "File URL path must not include encoded / characters"));
+
+      return false;
+    }
+  }
+
+  result_file_path = ada::unicode::percent_decode(pathname, pathname.size());
+
+  return true;
+#endif
+}
+
+FilePathIsFileReturnType FilePathIsFile(Environment* env,
+                                        const std::string& file_path) {
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env,
+      permission::PermissionScope::kFileSystemRead,
+      file_path,
+      FilePathIsFileReturnType::kThrowInsufficientPermissions);
+
+  uv_fs_t req;
+
+  int rc = uv_fs_stat(env->event_loop(), &req, file_path.c_str(), nullptr);
+
+  if (rc == 0) {
+    const uv_stat_t* const s = static_cast<const uv_stat_t*>(req.ptr);
+    rc = !!(s->st_mode & S_IFDIR);
+  }
+
+  uv_fs_req_cleanup(&req);
+
+  // rc is 0 if the path refers to a file
+  if (rc == 0) return FilePathIsFileReturnType::kIsFile;
+
+  return FilePathIsFileReturnType::kIsNotFile;
+}
+
+static void LegacyMainResolve(const FunctionCallbackInfo<Value>& args) {
+  CHECK_GE(args.Length(), 1);
+  // packageJSONUrl
+  // /home/user/project/package.json
+  CHECK(args[0]->IsString());
+
+  Environment* env = Environment::GetCurrent(args);
+
+  std::string_view base_path =
+      Utf8Value(env->isolate(), args[0].As<String>()).ToStringView();
+  auto base = ada::parse<ada::url_aggregator>(base_path);
```

Comment:
```suggestion
  Utf8Value base_path(env->isolate(), args[0].As<String>());
  auto base = ada::parse<ada::url_aggregator>(base_path.ToStringView());
```
The Utf8Value object needs to be on the stack for the returned string view to be valid as the actual string is owned by the Utf8Value object. This should fix the ASAN error you're facing.

Reply:
It worked, thanks!